### PR TITLE
setup project skeleton, config files, etc

### PR DESCRIPTION
**Pull request by chrisantonellis**
Added docker components, containerized lint + test suite:
* `docker/Dockerfile`
* `docker/run_tests.sh`
* `docker-compose.yaml`

Added root level config files to support lint, test suite:
* `.bandit`
* `.coveragerc`
* `.flake8`
* `mypy.ini`
* `pyproject.toml`
* `pytest.ini`
* `requirements.txt`
* `requirements-test.txt`

Added root level files to support packaging:
* `setup.py`
* `MANIFEST.in`

### CODE REVIEW

File: mypy.ini [hunk 1]
```diff
@@ -0,0 +1,2 @@
+[mypy]
+ignore_missing_imports = True
```
Comment (jamescurtin) [inline mypy.ini:2]: nit: trailing newline

```suggestion
ignore_missing_imports = True

```

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,26 @@
+[tool.black]
+line-length = 88
+target_version = ['py36']
```
Comment (jamescurtin) [inline pyproject.toml:3]: Based on the fact that this is being run in a Python 3.8 container
```suggestion
target_version = ['py38']
```

File: pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  # The following are specific to Black, you probably don't want those.
+  | blib2to3
+  | tests/data
+)/
+'''
+
+[tool.isort]
+line_length=88
+multi_line_output=3
+include_trailing_comma=true
```
Comment (jamescurtin) [inline pyproject.toml:26]: Using an [`isort` profile](https://pycqa.github.io/isort/docs/configuration/profiles/) simplifies the configuration & eliminates some weird edge cases where isort and black format differently.

```suggestion
profile = "black"
```

File: requirements-test.txt [hunk 4]
```diff
@@ -0,0 +1,9 @@
+bandit==1.7.0
+black==20.8b1
+flake8==3.8.4
+isort==5.7.0
+mypy==0.790
+pytest==6.2.1
+pytest-cov==2.10.1
+pytest-mock==3.4.0
+pdbpp
```
Comment (jamescurtin) [inline requirements-test.txt:9]: nit: could pin this as well

File: pytest.ini [hunk 5]
```diff
@@ -0,0 +1,3 @@
+[pytest]
+addopts = --verbose --cov=little_cheesemonger --cov-report xml:/tmp/coverage.xml --cov-report term-missing
```
Comment (jamescurtin) [inline pytest.ini:2]: Nit: could add an option to enforce minimum coverage percentage (I think this currently has 100% coverage because of the "hello world" test), but could also be added later.

**Review by jamescurtin — APPROVED** [review]
LGTM! You might also want to consider adding a boilerplate dependabot config as well as (optionally) issue templates. (This could be a followup PR, but wanted to highlight those two specifically because I didn't see them called out in the epic proposal)

File: pytest.ini (hunk 5 above)
Comment (chrisantonellis) [reply pytest.ini:2]: enforced here: https://github.com/wayfair-incubator/little-cheesemonger/pull/2/files#diff-834e9b406d74791ffbafaeec9cc894082cea9739bc347b6ff9f72312c92ebc79R10

😎
